packages/contracts-bedrock/src/L2/L2ERC721Bridge.sol

```diff
@@ -20,18 +20,20 @@ import { Predeploys } from "src/libraries/Predeploys.sol";
 ///         wait for the one-week challenge period to elapse before their Optimism-native NFT
 ///         can be refunded on L2.
 contract L2ERC721Bridge is ERC721Bridge, ISemver {
-    /// @custom:semver 1.6.0
-    string public constant version = "1.6.0";
+    /// @custom:semver 1.7.0
+    string public constant version = "1.7.0";
 
     /// @notice Constructs the L2ERC721Bridge contract.
-    /// @param _otherBridge Address of the ERC721 bridge on the other network.
-    constructor(address _otherBridge) ERC721Bridge(Predeploys.L2_CROSS_DOMAIN_MESSENGER, _otherBridge) {
-        initialize();
+    constructor(address _otherBridge) ERC721Bridge() {
+        initialize({ _messenger: CrossDomainMessenger(Predeploys.L2_CROSS_DOMAIN_MESSENGER), _otherBridge: _otherBridge });
     }
 
-    /// @notice Initializes the contract. This is a noop in the implementation but included to ensure that
-    ///         the contract cannot be initialized a second time.
-    function initialize() public initializer { }
+    /// @notice Initializes the contract.
+    /// @param _messenger   Contract of the CrossDomainMessenger on this network.
+    /// @param _otherBridge Address of the ERC721 bridge on the other network.
+    function initialize(CrossDomainMessenger _messenger, address _otherBridge) public initializer {
```

Comment:
We can hardcode the `CrossDomainMessenger` to be the predeploy address here and only pass `_otherBridge` as the argument